src/app/shared/components/template/services/template-calc.service.ts

```diff
@@ -41,10 +62,23 @@ export class TemplateCalcService extends AsyncServiceBase {
       this.calcContext = this.generateCalcContext();
     }
     // Assign all calc functions also to window object to allow calling between functions
-    (window as any).calc = this.calcFunctions;
+    this.windowWithCalc.calc = this.calcFunctions;
     return this.calcContext;
   }
 
+  /**
+   * Evaluate inner expression provided by `@calc(...)` expression
+   * The expression is evaluated as JS, with additional access to global constants, function and
+   * evaluation context variables
+   * */
+  public evaluate(expression: string, evalContext: FlowTypes.TemplateRowEvalContext = {}) {
+    const calcContext = this.getCalcContext();
+    const calcExpression = expression.replace(/@/gi, "this.");
+    const { thisCtxt, globalFunctions, globalConstants } = calcContext;
+    const mergedContext = { ...thisCtxt, ...evalContext };
+    return evaluateJSExpression(calcExpression, mergedContext, globalFunctions, globalConstants);
+  }
```

Comment:
Note - this method has been adapted from the template-variable service which currently handles evaluation. It will be used as part of the refactor in #2772